### PR TITLE
Update OkHttp to v3.6.0

### PR DESCRIPTION
**Pull request by aahlenst**
okhttp-ws is gone, see https://github.com/square/okhttp/blob/master/CHANGELOG.md#version-350. Without an update to at least OkHttp 3.5 it's very hard to use the java-sdk with newer versions of OkHttp, especially on Android.

I've tried to keep the changes to a minimum.

### CODE REVIEW

File: speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/websocket/WebSocketManager.java [hunk 1]
```diff
@@ -101,34 +98,36 @@
...
      *
-     * @see okhttp3.ws.WebSocketListener#onFailure(java.io.IOException, okhttp3.Response)
+     * @see okhttp3.WebSocketListener#onFailure(okhttp3.WebSocket, java.lang.Throwable, okhttp3.Response)
      */
     @Override
-    public void onFailure(IOException e, Response response) {
+    public void onFailure(WebSocket webSocket, Throwable t, Response response) {
       socketOpen = false;
-      callback.onError(e);
+      if (t instanceof Exception) {
```
Comment (aahlenst) [inline speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/websocket/WebSocketManager.java]: It's not pretty but I didn't want to break `RecognizeCallback`.

**Comment by germanattanasio** [comment]
@aahlenst Thanks for the PR, it will take some time for me to look at it since we are in the middle of migrating from maven to gradle.

Everything looks good after a quick look. Can you update the CHANGELOG describing the changes you made ?

**Comment by aahlenst** [comment]
Changelog is updated. Just tell me if it's not okay this way.

**Comment by germanattanasio** [comment]
This looks pretty good @aahlenst!

**Comment by troyibm** [comment]
When can this PR be delivered.  I have an app that I am trying to include speech-to-text in that uses OkHttp 3.6 and it doesn't build because the latest release of java-sdk includes older version of OkHttp.

**Comment by germanattanasio** [comment]
@troyibm we are trying to fix the deployment script. Right now we are having some issues with GPG but I hope to have it fixed by the end of the week

**Comment by troyibm** [comment]
thanks @germanattanasio.  Sorry to push.  I hit enter on my comment and then saw that you were working on this just yesterday. :)

One question, are you guys using gradle to build?  I forked Andreas' repo and did `./gradlew build` and got a testing failure in conversation:test.  I had to remove the @Test from ConversationTest.java to get build to pass.  But now I can't get it to build the watson-developer-cloud jar file.  `gradle jar` builds a 261 byte java-sdk.jar.

**Comment by germanattanasio** [comment]
@blakesteve can you look into the issue that @troyibm is mentioning. I think you have it working locally.

**Comment by troyibm** [comment]
@blakesteve if you need me to open an issue for the problems I am having with gradle, I can.

**Comment by troyibm** [comment]
@germanattanasio what's the ETA for 3.6.0 release of java-sdk?

**Comment by germanattanasio** [comment]
No ETA yet.

**Pull request merged**